src/services/objections.api.service.ts

```diff
@@ -1,52 +1,46 @@
-import logger from "../utils/logger";
-
-/**
- * State of Objection
- */
-export enum ObjectionStatus {
-  OPEN,
-  SUBMITTED,
-  PROCESSED,
-}
 
-/**
- * Data strucrure for patching an Objection
- */
-export interface ObjectionPatch {
-  reason?: string;
-  status?: ObjectionStatus;
-}
+import * as objectionsSdk from "../modules/sdk/objections";
+import logger from "../utils/logger";
 
 /**
  * Create a new objection for the given company.
  *
- * @param companyNumber the company number
- * @param token the bearer security token to use to call the api
+ * @param {string} companyNumber the company number
+ * @param {string} token the bearer security token to use to call the api
  *
- * @returns the id of the newly created objection
+ * @returns {string} the id of the newly created objection
+ * @throws {ApiError}
  */
-export const createNewObjection = (companyNumber: string, token: string): string => {
+export const createNewObjection = async (companyNumber: string, token: string): Promise<string> => {
+  logger.info(`Creating objection for company number ${companyNumber}`);
```

Comment:
I wonder if we need this class now? If all we have in it are a bunch of one-line functions that simply delegate to the sdk class functions it's arguably a pretty useless additional layer of code.

Reply:
The idea is to have an objections service that the controllers can interact with.
It also provides loose coupling with the sdk, a single point of contact. If all interactions are with the service then it minimises change if the sdk interface changes - we wouldn't have to update several controllers, only the service.

For example, I've split the patch request into 2 separate functions now, one to update the reason, other to update the status. There will have to be some patch object creation done in here (objections service). This will ensure that the SDK types such as Status types won't be polluting other code outside the service such as the controllers. Only the objections service should be exposed to the SDK code.

Reply:
Yep. Good argument.